tests/libgstc/test_libgstc_socket.c

```diff
@@ -85,6 +89,18 @@ mock_server_new ()
 
   _loop = g_main_loop_new (NULL, FALSE);
   _mock_thread = g_thread_new ("mock_server", mock_server_thread, NULL);
+
+  /* 
+   * FIXME this is done in order to ensure that the socket loop is running
+   * before starting the test and isn't dereferenced in an inconsistent state.
+   * This timeouts after 1s
+   */
+  start_time = g_get_monotonic_time ();
+  while (!g_main_loop_is_running (_loop) && (time_passed < start_timeout)) {
```

Comment:
We are not reacting to a timeout. Let's handle this timeout as a `g_return_if_fail` inside the while, so the test fails in the case of a timeout.